lib/model/Fixture.js

```diff
@@ -0,0 +1,173 @@
+const path = require('path');
+const fs = require('fs');
+
+const Meta = require(path.join(__dirname, 'Meta.js'));
+const Physical = require(path.join(__dirname, 'Physical.js'));
+const Channel = require(path.join(__dirname, 'Channel.js'));
+const Mode = require(path.join(__dirname, 'Mode.js'));
+
+module.exports = class Fixture {
+  constructor(jsonObject) {
+    this.jsonObject = jsonObject; // calls the setter
+  }
+
+  set jsonObject(jsonObject) {
+    this._jsonObject = jsonObject;
+    this._cache = {};
+  }
+
+
+  get name() {
+    return this._jsonObject.name; // required
+  }
+
+  get shortName() {
+    return this._jsonObject.shortName || this._jsonObject.name;
+  }
+
+  get categories() {
+    return this._jsonObject.categories; // required
+  }
+
+  get mainCategory() {
+    return this.categories[0];
+  }
+
+  get meta() {
+    if (!('meta' in this._cache)) {
+      this._cache.meta = new Meta(this._jsonObject.meta);
+    }
+
+    return this._cache.meta;
+  }
+
+  get comment() {
+    return this._jsonObject.comment || '';
+  }
+
+  get hasComment() {
+    return 'comment' in this._jsonObject;
+  }
+
+  get manualURL() {
+    return this._jsonObject.manualURL || null;
+  }
+
+  /*
+    benchmark results (10,000,000 iterations):
+    - without cache: ~1.9s
+    - with cache: ~0.52s (nearly 4 times faster!)
+    => that proves why caching, even for these small objects, is useful
+  */
+  get physical() {
+    if (!('physical' in this._cache)) {
+      this._cache.physical = new Physical(this._jsonObject.physical);
+    }
+
+    return this._cache.physical;
+  }
+
+  // array of all channel keys and aliases, ordered by appearance
+  get channelKeys() {
+    if (!('channelKeys' in this._cache)) {
+      this._cache.channelKeys = [];
+
+      for (const chKey of this.availableChannelKeys) {
+        this._cache.channelKeys.push(chKey);
+
+        const channel = this.availableChannels[chKey];
+        for (const alias of channel.fineChannelAliases) {
+          this._cache.channelKeys.push(alias);
+        }
+
+        for (const alias of Object.keys(channel.switchingChannels)) {
+          this._cache.channelKeys.push(alias);
+        }
+      }
+    }
+
+    return this._cache.channelKeys;
+  }
+
+  // ordered by appearance
+  // -> that makes it different from calling Object.keys(fix.availableChannels);
+  get availableChannelKeys() {
+    return Object.keys(this._jsonObject.availableChannels);
+  }
+
+  // { 'channel key': Channel, ... }
```

Comment:
This should return an array of `Channel`s. To find out a `Channel`'s key, we have its `key` property. To get an array with all channel keys, we have the `availableChannelKeys` getter (which will have to be rewritten).